stdlib/public/Concurrency/Task.swift

```diff
@@ -900,6 +900,106 @@ extension Task where Success == Never, Failure == Never {
+  }
+
+  /// Escalate the task `priority` of the passed in task to the `newPriority`.
+  ///
+  /// - Warning: This API should rarely be used, and instead you can rely on
+  ///   structured concurrency and implicit priority escalation which happens
+  ///   when a higher priority task awaits on a result of a lower priority task.
+  ///
+  ///   I.e. using `await` on the target task usually is the correct way to
+  ///   escalate the target task to the current priority of the calling task,
+  ///   especially because in such setup if the waiting task gets escalated,
+  ///   the waited on task would be escalated automatically as well.
+  ///
+  /// The concurrency runtime is free to interpret and handle escalation
+  /// depending on platform characteristics.
+  ///
+  /// Priority escalation is propagated to child tasks of the waited-on task,
+  /// and will trigger any priority escalation handlers, if any were registered.
+  ///
+  /// Escalation can only *increase* the priority of a task, and
+  /// de-escalating priority is not supported.
+  ///
+  /// This method can be called from any task or thread.
+  ///
+  /// - Parameters:
+  ///   - task: the task which to escalate the priority of
+  ///   - newPriority: the new priority the task should continue executing on
+  @available(SwiftStdlib 9999, *)
+  public static func escalatePriority(_ task: UnsafeCurrentTask, to newPriority: TaskPriority) {
+    _taskEscalate(task._task, newPriority: newPriority.rawValue)
+  }
+}
+
+/// Runs the passed `operation` while registering a task priority escalation handler.
+/// The handler will be triggered concurrently to the current task if the current
+/// is subject to priority escalation.
+///
+/// The handler may perform additional actions upon priority escalation,
+/// but cannot influence how the escalation influences the task, i.e. the task's
+/// priority will be escalated regardless of actions performed in the handler.
+///
+/// If multiple task escalation handlers are nester they will all be triggered.
+///
+/// Task escalation propagates through structured concurrency child-tasks.
+///
+/// - Parameters:
+///   - operation: the operation during which to listen for priority escalation
+///   - handler: handler to invoke, concurrently to `operation`,
+///              when priority escalation happens
+/// - Returns: the value returned by `operation`
+/// - Throws: when the `operation` throws an error
+@available(SwiftStdlib 9999, *)
+public func withTaskPriorityEscalationHandler<T, E>(
+  operation: () async throws(E) -> T,
+  onEscalate handler: @Sendable (TaskPriority) -> Void,
+  isolation: isolated (any Actor)? = #isolation
+) async throws(E) -> T {
+  // NOTE: We have to create the closure beforehand as otherwise it seems
+  // the task-local allocator may be used and we end up violating stack-discipline
+  // when releasing the handler closure vs. the record.
+  let handler0: (UInt8) -> Void = {
+    handler(TaskPriority(rawValue: $0))
+  }
```

Comment:
This is weird; we have to create the closure outside of the `_taskAddPriorityEscalationHandler` separately like this or we'll end up violating the task allocator's discipline; If there's a nicer way to express not using the task allocator for it that'd be good, perhaps some annotation on the `_taskAddPriorityEscalationHandler` that I'm missing?